OpenRA.Mods.Common/Traits/Production/Production.cs

```diff
@@ -37,48 +38,49 @@ public class Production
 		public Production(ActorInitializer init, ProductionInfo info)
 		{
 			Info = info;
-			rp = Exts.Lazy(() => init.Self.IsDead ? null : init.Self.TraitOrDefault<RallyPoint>());
+			RallyPoint = Exts.Lazy(() => init.Self.IsDead ? null : init.Self.TraitOrDefault<RallyPoint>());
 			Race = init.Contains<RaceInit>() ? init.Get<RaceInit, string>() : init.Self.Owner.Country.Race;
 		}
 
-		public void DoProduction(Actor self, ActorInfo producee, ExitInfo exitinfo, string raceVariant)
+		public virtual Actor DoProduction(Actor self, ActorInfo actorToProduce, ExitInfo exitInfo, string raceVariant)
 		{
-			var exit = self.Location + exitinfo.ExitCell;
-			var spawn = self.CenterPosition + exitinfo.SpawnOffset;
+			var exit = self.Location + exitInfo.ExitCell;
+			var spawn = self.CenterPosition + exitInfo.SpawnOffset;
 			var to = self.World.Map.CenterOfCell(exit);
 
-			var fi = producee.Traits.GetOrDefault<IFacingInfo>();
-			var initialFacing = exitinfo.Facing < 0 ? Util.GetFacing(to - spawn, fi == null ? 0 : fi.GetInitialFacing()) : exitinfo.Facing;
+			var fi = actorToProduce.Traits.GetOrDefault<IFacingInfo>();
+			var initialFacing = exitInfo.Facing < 0 ? Util.GetFacing(to - spawn, fi == null ? 0 : fi.GetInitialFacing()) : exitInfo.Facing;
 
-			var exitLocation = rp.Value != null ? rp.Value.Location : exit;
+			var exitLocation = RallyPoint.Value != null ? RallyPoint.Value.Location : exit;
 			var target = Target.FromCell(self.World, exitLocation);
 
-			var bi = producee.Traits.GetOrDefault<BuildableInfo>();
+			var bi = actorToProduce.Traits.GetOrDefault<BuildableInfo>();
 			if (bi != null && bi.ForceRace != null)
 				raceVariant = bi.ForceRace;
 
-			self.World.AddFrameEndTask(w =>
+			var td = new TypeDictionary
 			{
-				var td = new TypeDictionary
-				{
-					new OwnerInit(self.Owner),
-					new LocationInit(exit),
-					new CenterPositionInit(spawn),
-					new FacingInit(initialFacing)
-				};
+				new OwnerInit(self.Owner),
+				new LocationInit(exit),
+				new CenterPositionInit(spawn),
+				new FacingInit(initialFacing)
+			};
 
-				if (raceVariant != null)
-					td.Add(new RaceInit(raceVariant));
+			if (!string.IsNullOrEmpty(raceVariant))
+				td.Add(new RaceInit(raceVariant));
 
-				var newUnit = self.World.CreateActor(producee.Name, td);
+			var newUnit = self.World.CreateActor(false, actorToProduce.Name, td);
```

Comment:
The actors are created at the same moment as before. They are just not added to the world until at least the end of the frame, or until there is a free exit they can use.
They shouldn't be ticking if they aren't in the world, so if anything, this *should* make them tick later.